### PR TITLE
Update identifiers with missing "No" or "Std"

### PR DESCRIPTION
**Pull request by mico**
related #73

### CODE REVIEW

File: update_codes.yaml [hunk 1]
```diff
@@ -45,3 +45,8 @@ P1484.12.3/D1, September2019: IEEE P1484.12.3/D1, September 2019
 P1073.1.3.4/D3.0: IEEE P1073.1.3.4/D3.0
 P1936.1, 2020: IEEE P1936.1, January 2020
 P20000-2, D1 December 2012: IEEE P20000-2/D1, December 2012
+IEEE 1070-1995: IEEE No 1070-1995
```
Comment (ronaldtse) [inline update_codes.yaml:48]: I disagree with this. We should not need to change “IEEE xxx” to “IEEE No xxx”.

File: update_codes.yaml [hunk 2]
```diff
@@ -45,3 +45,8 @@ P1484.12.3/D1, September2019: IEEE P1484.12.3/D1, September 2019
...
+IEEE 278-1967: IEEE No 278-1967
+IEEE 730-1989: IEEE No 730-1989
+IEEE 751-1991: IEEE No 751-1991
+IEEE 1023-2020 (Revision of IEEE Std 1023-2004): IEEE No 1023-2020 (Revision of IEEE Std 1023-2004)
```
Comment (ronaldtse) [inline update_codes.yaml:52]: Also disagree. Thanks.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Comment by ronaldtse** [comment]
Can’t this just be done by making “Std” and “No” by optional?

File: spec/pubid_ieee/identifier_spec.rb [hunk 3]
```diff
@@ -824,6 +824,14 @@ def context_description(ex)
 
       it_behaves_like "converts pubid to pubid"
     end
+
+    context "IEEE 278-1967" do
+      let(:pubid) { "IEEE 278-1967" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:829]: Can we also set the generated PubID to be "IEEE Std 278-1967"?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Comment by ronaldtse** [comment]
@mico can you help update/rebase this PR and merge it? Thanks!

**Comment by mico** [comment]
@ronaldtse added "Std" type by default. Could you please confirm if it's what you meant?

File: spec/pubid_ieee/identifier_spec.rb [hunk 4]
```diff
@@ -553,14 +556,14 @@ def context_description(ex)
...
-      let(:pubid) { "ANSI C50.32-1976 (IEEE Std 117-1974) (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" }
+      let(:pubid) { "ANSI Std C50.32-1976 (IEEE Std 117-1974) (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" }
 
       it_behaves_like "converts pubid to pubid"
     end
 
     context "IEEE No 29-1941 / ASA C77.1-1943 (Reaffirmed 1971)" do
       let(:original) { "IEEE No 29-1941 / ASA C77.1-1943 (Reaffirmed 1971)" }
-      let(:pubid) { "IEEE 29-1941 (ASA C77.1-1943) (Reaffirmed 1971)" }
+      let(:pubid) { "IEEE Std 29-1941 (ASA Std C77.1-1943) (Reaffirmed 1971)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:566]: For "ASA C77.1-1943" we don't want the "Std".

File: spec/pubid_ieee/identifier_spec.rb [hunk 5]
```diff
@@ -553,14 +556,14 @@ def context_description(ex)
 
     context "ANSI C50.32-1976 and IEEE Std 117-1974 (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" do
       let(:original) { "ANSI C50.32-1976 and IEEE Std 117-1974 (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" }
-      let(:pubid) { "ANSI C50.32-1976 (IEEE Std 117-1974) (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" }
+      let(:pubid) { "ANSI Std C50.32-1976 (IEEE Std 117-1974) (Reaffirmed 1984) (Revision of IEEE Std 117-1956)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:559]: For "ANSI C50.32-1976" we don't want the "Std" added.

File: spec/pubid_ieee/identifier_spec.rb [hunk 6]
```diff
@@ -539,7 +541,8 @@ def context_description(ex)
     end
 
     context "ASA C37.1-1937 (Reaffirmed 1945)" do
-      let(:pubid) { "ASA C37.1-1937 (Reaffirmed 1945)" }
+      let(:original) { "ASA C37.1-1937 (Reaffirmed 1945)" }
+      let(:pubid) { "ASA Std C37.1-1937 (Reaffirmed 1945)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:545]: No "Std" here please. Only for "IEEE" prefix.

File: spec/pubid_ieee/identifier_spec.rb [hunk 7]
```diff
@@ -411,7 +413,7 @@ def context_description(ex)
 
     context "IEEE P802.11ajD8.0, August 2017" do
       let(:original) { "IEEE P802.11ajD8.0, August 2017" }
-      let(:pubid) { "IEEE P802.11aj/D8.0, August 2017" }
+      let(:pubid) { "IEEE Std P802.11aj/D8.0, August 2017" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:416]: The "Pxxxx" ones are not published standards -- they are preliminary projects, so they should not have "Std".

File: spec/pubid_ieee/identifier_spec.rb [hunk 8]
```diff
@@ -24,7 +24,7 @@ def context_description(ex)
   describe "parse specific identifiers" do
     context "IEEE No 142-1956" do
       let(:original) { "IEEE No 142-1956" }
-      let(:pubid) { "IEEE 142-1956" }
+      let(:pubid) { "IEEE Std 142-1956" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:27]: This is correct.

File: spec/pubid_ieee/identifier_spec.rb [hunk 9]
```diff
@@ -50,7 +50,8 @@ def context_description(ex)
     end
 
     context "ANSI C37.0781-1972" do
-      let(:pubid) { "ANSI C37.0781-1972" }
+      let(:original) { "ANSI C37.0781-1972" }
+      let(:pubid) { "ANSI Std C37.0781-1972" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:54]: For ANSI we don't add Std.

File: spec/pubid_ieee/identifier_spec.rb [hunk 10]
```diff
@@ -62,34 +63,35 @@ def context_description(ex)
     end
 
     context "ASA C37.1-1950" do
-      let(:pubid) { "ASA C37.1-1950" }
+      let(:original) { "ASA C37.1-1950" }
+      let(:pubid) { "ASA Std C37.1-1950" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:67]: For ASA we don't add "Std".

File: spec/pubid_ieee/identifier_spec.rb [hunk 11]
```diff
@@ -62,34 +63,35 @@ def context_description(ex)
...
-      let(:pubid) { "ANSI/IEEE C37.23-1969" }
+      let(:pubid) { "ANSI/IEEE Std C37.23-1969" }
 
       it_behaves_like "converts pubid to pubid"
     end
 
     context "AIEE No 91-1962 (ASA Y32.14-1962)" do
       let(:original) { "AIEE No 91-1962 (ASA Y32.14-1962)" }
-      let(:pubid) { "AIEE 91-1962 (ASA Y32.14-1962)" }
+      let(:pubid) { "AIEE Std 91-1962 (ASA Std Y32.14-1962)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:81]: AIEE Standards are old, and we should not add "Std".

File: spec/pubid_ieee/identifier_spec.rb [hunk 12]
```diff
@@ -62,34 +63,35 @@ def context_description(ex)
...
       let(:original) { "ANSI C37.61-1973 and IEEE Std 321-1973" }
-      let(:pubid) { "ANSI C37.61-1973 (IEEE Std 321-1973)" }
+      let(:pubid) { "ANSI Std C37.61-1973 (IEEE Std 321-1973)" }
 
       it_behaves_like "converts pubid to pubid"
     end
 
     context "IEEE Std 623-1976 (ANSI Y32.21-1976, NCTA 006-0975)" do
-      let(:pubid) { "IEEE Std 623-1976 (ANSI Y32.21-1976, NCTA 006.0975)" }
+      let(:pubid) { "IEEE Std 623-1976 (ANSI Std Y32.21-1976, NCTA Std 006.0975)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:94]: For ANSI and NCTA we don't add Std.

File: spec/pubid_ieee/identifier_spec.rb [hunk 13]
```diff
@@ -145,21 +147,21 @@ def context_description(ex)
...
+      let(:pubid) { "ANSI Std PC63.10/D14, April 2020" }
+      let(:full_pubid) { "ANSI Draft Std PC63.10/D14, April 2020" }
 
       it_behaves_like "converts pubid to pubid"
     end
 
     context "IEEE Std 1076.1 IEC 61691-6 Edition 1.0 2009-12" do
       let(:original) { "IEEE Std 1076.1 IEC 61691-6 Edition 1.0 2009-12" }
-      let(:pubid) { "IEEE Std 1076.1 (IEC 61691-6 Edition 1.0 2009-12)" }
+      let(:pubid) { "IEEE Std 1076.1 (IEC Std 61691-6 Edition 1.0 2009-12)" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:164]: IEC should not have Std.

File: spec/pubid_ieee/identifier_spec.rb [hunk 14]
```diff
@@ -179,19 +181,19 @@ def context_description(ex)
...
     context "ANSI PC63.12/D12e, January 2015" do
-      let(:pubid) { "ANSI PC63.12/D12e, January 2015" }
+      let(:pubid) { "ANSI Std PC63.12/D12e, January 2015" }
 
       it_behaves_like "converts pubid to pubid"
     end
 
     context "IEEE P62.44/D15.3" do
-      let(:pubid) { "IEEE P62.44/D15.3" }
+      let(:pubid) { "IEEE Std P62.44/D15.3" }
```
Comment (ronaldtse) [inline spec/pubid_ieee/identifier_spec.rb:190]: "Pxxx" do not add Std because they are not yet published.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Comment by ronaldtse** [comment]
> @ronaldtse added "Std" type by default. Could you please confirm if it's what you meant?

Yes that's correct, but only for "IEEE" published documents:
* For other prefixes, ASA, AIEE, ANSI do not add "Std"
* For "IEEE Pxxx" documents they are preliminary projects -- not yet published -- hence not "Standards" and should not have "Std".

**Review by ronaldtse — APPROVED** [review]
Very nice. Thank you @mico !

**Comment by ronaldtse** [comment]
Very nicely done. Thank you @mico !

**Pull request merged**